### PR TITLE
Remove image() macro (Fixes #12190)

### LINKED ISSUE #12190: Remove image macro
As noted in the[ card highres image](https://github.com/mozilla/bedrock/pull/12169#pullrequestreview-1128615261) fix, the [image macro](https://github.com/mozilla/bedrock/blob/main/bedrock/base/templates/macros.html#L143) is very complex. It was created to encapsulate shared complexity across several protocol macros using images.

We need to handle:
- static images
- localized images
- high res images
- images with multiple sources (picture elements, entirely different images or image formats)
- images with multiple sizes

We also have some python helpers for:
- resp_img
- picture
- high_res_img (deprecated)

In most cases, resp_img should cover what we need to handle. We should investigate if we can remove the image macro and simplify our protocol macro image calls.

Success Criteria:

- [x] Protocol macro use of image macro is replaced
- [x] Standalone use of image macro is replaced
- [x] Image macro and related docs/tests are removed

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

- Removes `image()` macro from bedrock and replaces with `res_img()`.
- Also tidies up a few stray hero component references.

## Issue / Bugzilla link

#12190

## Testing

- [x] http://localhost:8000/en-US/firefox/pocket/
- [x] http://localhost:8000/en-US/firefox/browsers/windows-64-bit/
- [x] http://localhost:8000/en-US/firefox/browsers/mobile/?xv=legacy
- [x] http://localhost:8000/en-US/firefox/new/?xv=basic
- [x] http://localhost:8000/en-US/firefox/set-as-default/thanks/
- [x] http://localhost:8000/en-US/foundation/annualreport/2019/
- [x] http://localhost:8000/en-US/foundation/annualreport/2020/
- [x] http://localhost:8000/en-US/foundation/annualreport/2021/
- [x] http://localhost:8000/en-US/diversity/2021/beyond-our-products/
- [x] http://localhost:8000/en-US/diversity/2021/
- [x] http://localhost:8000/en-US/diversity/2021/what-we-build/
- [x] http://localhost:8000/en-US/diversity/2021/who-we-are/
- [x] http://localhost:8000/en-US/stories/
- [x] http://localhost:8000/en-US/stories/joy-of-color/
- [x] http://localhost:8000/en-US/about/policy/lean-data/

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/759650900

### CODE REVIEW

File: bedrock/careers/templates/careers/internships.html [hunk 1]
```diff
@@ -12,14 +12,14 @@
 {% block body_class %}internships-page{% endblock %}
 
 {% block careers_content %}
-<section class="mzp-l-content mzp-c-hero mzp-t-dark">
```
Comment (alexgibson) [inline bedrock/careers/templates/careers/internships.html:15]: Note for reviewer. This page is temporarily being redirected back to the /careers home page. I'm not sure if it's still needed, but appears to be in a somewhat half finished state perhaps (I'm not too sure).

I had to remove the redirect locally and define a URL in order to view the changes here.

File: bedrock/contentful/api.py [hunk 2]
```diff
@@ -692,29 +686,6 @@ def get_text_data(self, value):
 
         return data
 
-    def get_hero_data(self, entry_obj):
```
Comment (alexgibson) [inline bedrock/contentful/api.py:695]: Note for reviewer: I'm assuming this is now safe to remove?

**Comment by reemhamz** [comment]
Just noticed @maureenlholland  assigned herself this ticket, and I didn't have a chance to catch up to this today, so I'll leave this to her if she has the time :) If not, I'll be happy to review when I'm back tomorrow.

**Review by maureenlholland — APPROVED** [review]
LGTM r+ 🧹 

Thanks for the test run! 

One pre-existing issue we could clean up here or in a separate PR

File: bedrock/careers/templates/careers/internships.html (hunk 1 above)
Comment (maureenlholland) [reply bedrock/careers/templates/careers/internships.html:15]: @reemhamz probably has the best insight here

**Pull request merged**